subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskStateInternal.java

```diff
@@ -122,4 +123,12 @@ public boolean getNoSource() {
     public boolean isFromCache() {
         return outcome == TaskExecutionOutcome.FROM_CACHE;
     }
+
+    public boolean isActionsWereExecuted() {
```

Comment:
I think it's fine as is... it just makes me read it very carefully.  And that's not all bad.